src/bun.js/api/bun/socket.zig

```diff
@@ -1706,17 +1798,26 @@ fn NewSocket(comptime ssl: bool) type {
 
         pub fn finalize(this: *This) callconv(.C) void {
             log("finalize()", .{});
-            if (this.detached) return;
-            this.detached = true;
-            if (!this.socket.isClosed()) {
-                this.socket.close(0, null);
+            if (!this.detached) {
+                this.detached = true;
+                if (!this.socket.isClosed()) {
+                    this.socket.close(0, null);
+                }
+                this.markInactive();
+                this.poll_ref.unref(JSC.VirtualMachine.get());
```

Comment:
feels safer to always .unref() since it becomes a no-op if unused, but fine to not do that

Reply:
fix in https://github.com/oven-sh/bun/pull/3457/commits/68b783e0bbb8c6721f291fbbfae114197b550491